background.js

```diff
@@ -1,128 +1,374 @@
-*/ 
-function initStorage(key, initialValue) {
-  var currentValue = localStorage[key];
-  if (!currentValue) {
-	localStorage[key] = initialValue;
-	return true;
-  }
-  return false;
-}
-
-rcxMain.config = {};
-rcxMain.config.css = localStorage["popupcolor"];
-rcxMain.config.highlight = localStorage["highlight"];
-rcxMain.config.textboxhl = localStorage["textboxhl"];
-rcxMain.config.onlyreading = localStorage["onlyreading"];
-rcxMain.config.copySeparator = localStorage["copySeparator"];
-rcxMain.config.maxClipCopyEntries = localStorage["maxClipCopyEntries"];
-rcxMain.config.lineEnding = localStorage["lineEnding"];
-rcxMain.config.minihelp = localStorage["minihelp"];
-rcxMain.config.popupDelay = parseInt(localStorage["popupDelay"]);
-rcxMain.config.disablekeys = localStorage["disablekeys"];
-rcxMain.config.showOnKey = localStorage["showOnKey"];
-rcxMain.config.kanjicomponents = localStorage["kanjicomponents"];
-rcxMain.config.kanjiinfo = new Array(rcxDict.prototype.numList.length/2);
-for (i = 0; i*2 < rcxDict.prototype.numList.length; i++) {
-	rcxMain.config.kanjiinfo[i] = localStorage[rcxDict.prototype.numList[i*2]];
-}
-
+/**
+ * Version info
+ *
+ */
+
+chrome.browserAction.onClicked.addListener(rcxMain.inlineToggle);
+chrome.tabs.onSelectionChanged.addListener(rcxMain.onTabSelect);
+chrome.runtime.onMessage.addListener(
+	function (request, sender, response) {
+		switch (request.type) {
+			case 'enable?':
+				console.log('enable?');
+				rcxMain.onTabSelect(sender.tab.id);
+				break;
+			case 'xsearch':
+				console.log('xsearch');
+				var e = rcxMain.search(request.text, request.dictOption);
+				response(e);
+				break;
+			/**  case 'nextDict':
+			 console.log('nextDict');
+			 rcxMain.nextDict();
+			 break;*/
+			case 'resetDict':
+				console.log('resetDict');
+				rcxMain.resetDict();
+				break;
+			case 'translate':
+				console.log('translate');
+				var e = rcxMain.dict.translate(request.title);
+				response(e);
+				break;
+			case 'makehtml':
+				console.log('makehtml');
+				var html = rcxMain.dict.makeHtml(request.entry);
+				response(html);
+				break;
+			case 'switchOnlyReading':
+				console.log('switchOnlyReading');
+				if (rcxMain.config.onlyreading == 'true')
+					rcxMain.config.onlyreading = 'false';
+				else
+					rcxMain.config.onlyreading = 'true';
+				//Todo: Replace this with cloud storage
+				localStorage['onlyreading'] = rcxMain.config.onlyreading;
+				break;
+			case 'copyToClip':
+				console.log('copyToClip');
+				rcxMain.copyToClip(sender.tab, request.entry);
+				break;
+			default:
+				console.log(request);
+		}
+	})
+
+var optionsList = [
+	"copySeparator", // Is this field separator?
+	"disablekeys",
+	"highlight",
+	"kanjicomponents",
+	"kanjiInfo",
+	"lineEnding",
+	"maxClipCopyEntries",
+	"minihelp",
+	"onlyreading",
+	"popupcolor",
+	"popupDelay",
+	"popupLocation",
+	"textboxhl",
+	"showOnKey"];
+
+var kanjiInfoLabelList = rcxDict.prototype.kanjiInfoLabelList;
```

Comment:
this initialization section changed a lot more than I thought it would so I'll look more later.

Looks like it may be caused by diffs poor move detection but the file is still double the number of lines...